src/lib/rest/StringFilter.cpp

```diff
@@ -1307,7 +1673,18 @@ bool StringFilter::match(ContextElementResponse* cerP)
       break;
 
     case SfopMatchPattern:
-      return itemP->matchPattern(caP);
+      //
+      // To not be forgotten for CNR (and discussed during PR-review):
+      //   This here is a bug fix.
+      //   If a 'q/mq' contains two expressions, a 'pattern' and an 'eq', and the eq gives false,
+      //   while the pattern gives true, then the total outcome will be true (before the bug fix),
+      //   because of the inconditional return that the bugfix removes.
+      //   The items in 'q/mq' are ANDed and as true AND false gives false, this was a bug.
```

Comment:
Now that we have discussed the above and added the entry to CNR I understand that the comment could be removed